pkg/distributor/ratestore_metrics.go

```diff
@@ -0,0 +1,50 @@
+package distributor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/instrument"
+)
+
+type ratestoreMetrics struct {
+	rateRefreshFailures *prometheus.CounterVec
+	streamCount         prometheus.Gauge
+	maxStreamShardCount prometheus.Gauge
+	maxStreamRate       prometheus.Gauge
+	refreshDuration     *instrument.HistogramCollector
+}
+
+func newRateStoreMetrics(reg prometheus.Registerer) *ratestoreMetrics {
+	return &ratestoreMetrics{
+		rateRefreshFailures: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Namespace: "loki",
+			Name:      "rate_store_refresh_failures_total",
+			Help:      "The total number of failed attempts to refresh the distributor's view of stream rates",
+		}, []string{"source"}),
+		streamCount: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Namespace: "loki",
+			Name:      "rate_store_stream_count",
+			Help:      "The last seen number of streams",
```

Comment:
This metric isn't super clear for me: is it the last seen number of streams by this rate store in a getRates call? or in a sync iteration?